aleph/worker.py

```diff
@@ -32,14 +33,25 @@
 from aleph.logic.entities import update_entity, prune_entity
 from aleph.logic.mapping import load_mapping, flush_mapping
 
+INDEXING_TIMEOUT = 10  # run all available indexing jobs in a batch after 10 seconds
+
 log = structlog.get_logger(__name__)
 
 app = create_app(config={"SERVER_NAME": settings.APP_UI_URL})
 
 
-def op_index(collection, task):
-    sync = task.context.get("sync", False)
-    index_many(task.operation, collection, sync=sync, **task.payload)
+def op_index(collection_id, batch, worker=None):
+    collection = Collection.by_id(collection_id)
```

Comment:
Are you still ensuring that all the documents in the whole batch are from the same collection? I would consider sending the collection as a dict as part of the indexing requests pretty soon after making this change, I think it'll be a perf game changer...

Reply:
@pudo So we want to batch indexing tasks irrespective of collection?